compiler/semantic/resolver.go

```diff
@@ -4,31 +4,40 @@ import (
 	"fmt"
 	"strconv"
 
+	"github.com/brimdata/super"
 	"github.com/brimdata/super/compiler/ast"
-	"github.com/brimdata/super/compiler/dag"
+	"github.com/brimdata/super/compiler/semantic/sem"
 	"github.com/brimdata/super/runtime/sam/expr/function"
 )
 
 type resolver struct {
-	analyzer *analyzer
-	in       map[string]*dag.FuncDef
-	fixed    map[string]*dag.FuncDef
-	variants []*dag.FuncDef
+	reporter
+	in       map[string]*sem.FuncDef
+	fixed    map[string]*sem.FuncDef
+	variants []*sem.FuncDef
 	params   []map[string]string
 	ntag     int
 }
 
-func newResolver(a *analyzer) *resolver {
-	r := &resolver{
-		analyzer: a,
-		in:       a.funcsByTag,
-		fixed:    make(map[string]*dag.FuncDef),
+func newResolver(r reporter, funcs map[string]*sem.FuncDef) *resolver {
+	return &resolver{
+		reporter: r,
+		in:       funcs,
+		fixed:    make(map[string]*sem.FuncDef),
 	}
-	return r
 }
 
-func (r *resolver) resolve(seq dag.Seq) (dag.Seq, []*dag.FuncDef) {
-	out := r.resolveSeq(seq)
+func (r *resolver) resolve(seq sem.Seq) (sem.Seq, []*sem.FuncDef) {
+	out := r.seq(seq)
+	funcs := r.variants
+	for _, f := range r.fixed {
+		funcs = append(funcs, f)
+	}
+	return out, funcs
+}
```

Comment:
Nit: This can reuse r.variant's underlying array across calls. Clearly that isn't a problem right now but you might want to future-proof with `funcs := slices.Clone(r.variants)`.